packages/util-dns/src/HostResolver.ts

```diff
@@ -0,0 +1,36 @@
+import { HostAddress, HostResolver as IHostResolver, HostResolverArguments } from "@aws-sdk/types";
+
+import { resolveAddress } from "./resolveAddress";
```

Comment:
Need feedback on a cleaner implementation (hacky by binding `resolveAddress()` to `this`).

Maybe it would be better to have per-file class implementations of `HostResolver`, e.g. one for the browser, one for `dns.lookup()`, one for `dns.resolve*()`.